pkg/router/template/util/haproxytime/duration_parser.go

```diff
@@ -0,0 +1,75 @@
+package haproxytime
+
+import (
+	"errors"
+	"math"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+var (
+	// OverflowError represents an overflow error from ParseDuration.
+	// OverflowError is returned if the value is greater than what time.ParseDuration
```

Comment:
This implementation doesn't use `time.ParseDuration`.

Reply:
ok